src/main/java/build/buildfarm/cas/CASFileCache.java

```diff
@@ -2189,34 +2201,41 @@ Digest getDigest() {
             },
             service);
 
-    ListenableFuture<Void> chmodFuture =
+    ListenableFuture<Void> chmodAndIndexFuture =
         transformAsync(
             fetchFuture,
             (result) -> {
               ImmutableList.Builder<Throwable> failures = ImmutableList.builder();
               boolean failed = false;
               try {
                 disableAllWriteAccess(path);
-              } catch (IOException chmodException) {
-                logger.log(
-                    Level.SEVERE,
-                    "error while disabling write permissions on " + path,
-                    chmodException);
-                return immediateFailedFuture(chmodException);
+              } catch (IOException e) {
+                logger.log(Level.SEVERE, "error while disabling write permissions on " + path, e);
+                return immediateFailedFuture(e);
```

Comment:
same as above: if all results are "immediate" you should be able to reuse a regular non-async transform()